src/sksevrhooks.h

```diff
@@ -5,22 +5,244 @@
+			intfc->ReadRecordData(&nameLen, sizeof(nameLen));
+			intfc->ReadRecordData(&name, nameLen);
+			name[nameLen] = 0;
+
+			std::uint32_t lightIndex = 0xFE000 | i;
+
+			const TESFile* file = dhand->LookupModByName(name);
+			if (file) {
+				std::uint32_t newIndex = file->GetPartialIndex();
+				s_savedModIndexMap[lightIndex] = newIndex;
+				logger::info("\t({} -> {})\t{}", lightIndex, newIndex, name);
+			} else {
+				s_savedModIndexMap[lightIndex] = 0xFF;
+			}
+			logger::info("\t({} -> {})\t{}", lightIndex, s_savedModIndexMap[lightIndex], name);
+		}
+	}
+
+	//FUN_180028550 + 0x1cb LoadModList // 2871b
+	void LoadModList(SKSE::SerializationInterface* intfc)
+	{
+		logger::info("Loading mod list:");
+
+		DataHandler* dhand = DataHandler::GetSingleton();
+
+		char name[0x104] = { 0 };
+		std::uint16_t nameLen = 0;
+
+		std::uint8_t numSavedMods = 0;
+		intfc->ReadRecordData(&numSavedMods, sizeof(numSavedMods));
+		for (std::uint32_t i = 0; i < numSavedMods; i++) {
+			intfc->ReadRecordData(&nameLen, sizeof(nameLen));
+			intfc->ReadRecordData(&name, nameLen);
+			name[nameLen] = 0;
+
+			const TESFile* modInfo = dhand->LookupModByName(name);
+			if (modInfo) {
+				std::uint32_t newIndex = modInfo->GetPartialIndex();
+				s_savedModIndexMap[i] = newIndex;
+				logger::info("\t({} -> {})\t{}", i, newIndex, name);
+			} else {
+				s_savedModIndexMap[i] = 0xFF;
+			}
+		}
+	}
+
+	//This is new for skse64 and should be triggered on 'PLGN':
+	//Attempt to inject at SKSEVR::FUN_180028550 + 0x30 before the old case statement.
+	void LoadPluginList(SKSE::SerializationInterface* intfc)
+	{
+		DataHandler* dhand = DataHandler::GetSingleton();
+
+		logger::info("Loading plugin list:");
+
+		char name[0x104] = { 0 };
+		std::uint16_t nameLen = 0;
+
+		std::uint16_t modCount = 0;
+		intfc->ReadRecordData(&modCount, sizeof(modCount));
+		for (std::uint32_t i = 0; i < modCount; i++) {
+			std::uint8_t modIndex = 0xFF;
+			std::uint16_t lightModIndex = 0xFFFF;
+			intfc->ReadRecordData(&modIndex, sizeof(modIndex));
+			if (modIndex == 0xFE) {
+				intfc->ReadRecordData(&lightModIndex, sizeof(lightModIndex));
+			}
+
+			intfc->ReadRecordData(&nameLen, sizeof(nameLen));
+			intfc->ReadRecordData(&name, nameLen);
+			name[nameLen] = 0;
+
+			std::uint32_t newIndex = 0xFF;
+			std::uint32_t oldIndex = modIndex == 0xFE ? (0xFE000 | lightModIndex) : modIndex;
+
+			const TESFile* modInfo = dhand->LookupModByName(name);
+			if (modInfo) {
+				newIndex = modInfo->GetPartialIndex();
+			}
+
+			s_savedModIndexMap[oldIndex] = newIndex;
+
+			logger::info("\t({} -> {})\t{}", oldIndex, newIndex, name);
+		}
+	}
+
+	struct Core_LoadCallback_Switch : Xbyak::CodeGenerator
+	{
+		Core_LoadCallback_Switch(std::uintptr_t beginSwitch, std::uintptr_t endLoop)
+		{
+			mov(rdx, dword[rsp + 0x8]);
+			cmp(rdx, 0x504C474E);  // 'PLGN'
+			jnz("KeepChecking");
+			mov(rcx, rbx);  // LoadPluginList(intfc);
+			call(&LoadPluginList);
```

Comment:
Suggestion: Same as https://github.com/Nightfallstorm/SkyrimVRESL/pull/36/files#r1424156503, `LoadPluginList` alone is considered a pointer to the function. Referencing it like this means it won't directly call the function, but the pointer to the pointer of the function, which ends up being in the wrong spot